modules/gatsby/src/gatsby-node/create-pages.js

```diff
@@ -67,6 +67,41 @@ function createSearchPage({ graphql, actions }) {
   );
 }
 
+function getExampleThumbnails({ allFile, allImageSharp }) {
+  // this function associates the path of an original image
+  // with the public url of a thumbnail, resized server side by imageSharp.
+  if (
+    !allFile ||
+    !allFile.edges ||
+    !allFile.edges.length ||
+    !allImageSharp ||
+    !allImageSharp.edges ||
+    !allImageSharp.edges.length
+  ) {
+    return {};
```

Comment:
Emit some warning?

Reply:
ok. 
There is something else - if we go down have a query that says 
```
siteMetadata {
          config {
            EXAMPLES {
              image
              title
              path
            }
          }
        }
      }
```

and there is at least one example THEN that example MUST have the fields image, title and path. If there is no example with the image field, that query will result in an error. 

So we could either - 

do a try / catch pattern with this query, and if it fails, do the same query without image and without looking for thumbnails, 

or enforce having an image field in all examples.

Reply:
Yes... for now I think it could be enough to have a minimal try catch that logs a message hinting at what went wrong.